File: signer/src/env.go
package signer

import (
	"fmt"
	"os"
)

type Env struct {
    HolderName, ServiceAddress string
}

func ReadEnv() (*Env, error) {
    result := Env{}

	result.HolderName = os.Getenv("HOLDER_NAME")
	if result.HolderName == "" {
	    return nil, fmt.Errorf("$HOLDER_NAME environment variable is missing or undefined")
	}

	result.ServiceAddress = os.Getenv("SERVICE_ADDRESS")
	if result.ServiceAddress == "" {
	    return nil, fmt.Errorf("$SERVICE_ADDRESS environment variable is missing or undefined")
	}

	return &result, nil
}

File: frontend/src/env.ts
export type Env = {
  VITE_API_URL: string,
  VITE_OWNER_NAME: string,
};

export const GetEnv = (): Env => {
  let result: Record<string, string> = {};
  for (let varname of ["VITE_API_URL", "VITE_OWNER_NAME"]) {
    result[varname] = import.meta.env[varname];
    if (result[varname] === undefined) {
      throw new Error(`Missing $${varname} env variable`);
    }
  }
  return result as Env;
};
